swap/tests/bob_xmr_validation.rs

```diff
@@ -0,0 +1,164 @@
+pub mod harness;
+
+use harness::FastCancelConfig;
+use swap::asb::FixedRate;
+use swap::protocol::alice::AliceState;
+use swap::protocol::bob::BobState;
+use swap::protocol::{alice, bob};
+use swap_core::monero::Amount;
+
+use crate::harness::SlowCancelConfig;
+
+#[tokio::test]
+async fn given_alice_locks_wrong_xmr_amount_bob_rejects() {
+    harness::setup_test(SlowCancelConfig, None, |mut ctx| async move {
+        // Run Bob until he gives up on the swap
+        let (bob_swap, bob_join_handle) = ctx.bob_swap().await;
+        let bob_swap_id = bob_swap.id;
+        let bob_swap = tokio::spawn(bob::run_until(bob_swap, |s| {
+            matches!(s, BobState::WaitingForCancelTimelockExpiration { .. })
+        }));
+
+        // Run until Alice detects the Bitcoin as having been locked
+        let alice_swap = ctx.alice_next_swap().await;
+
+        let alice_state = alice::run_until(
+            alice_swap,
+            |s| matches!(s, AliceState::BtcLocked { .. }),
+            FixedRate::default(),
+        )
+        .await?;
+
+        // Resume Alice such that she locks the wrong amount of Monero
+        ctx.restart_alice().await;
+        let mut alice_swap = ctx.alice_next_swap().await;
+
+        // Modify Alice such that she locks the wrong amount of Monero
+        alice_swap.state = alice_state.lower_by_one_piconero();
+        let alice_swap = tokio::spawn(alice::run(alice_swap, FixedRate::default()));
+
+        // Run until Bob detects the wrong amount of Monero and gives up
+        // He gives up by waiting for the cancel timelock to expire
+        let bob_state = bob_swap.await??;
+        assert!(matches!(
+            bob_state,
+            BobState::WaitingForCancelTimelockExpiration { .. }
```

Comment:
Same here: We need to ensure that Bob goes into `XmrLockTransactionCandidate` and then into `WaitingForCancelTimelockExpiration`.